.github/workflows/upgrade.yaml

```diff
@@ -0,0 +1,64 @@
+name: Upgrade Juju Controller and Juju Model
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Runs everyday at 00:00. (see https://crontab.guru)
+    - cron: "0 0 * * *"
```

Comment:
upgrading the controller shouldn't be a recurrent event, Additionally, the ``AGENT_VERSION`` should be a workflow parameter, not a secret. Similarly to how you'd pass an argument to a function.